### PR TITLE
CRM-18563 Pass expected error message to assertAPIFailure

### PR DESCRIPTION
**Pull request by mollux**
- [CRM-18563: Issues with the API and testing the API](https://issues.civicrm.org/jira/browse/CRM-18563)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
looks good - but seems to be flushing out a few things that need to be fixed before it will pass...

**Comment by mollux** [comment]
Hurray tests pass :)
the most important change is splitting up the _civicrm_api3_validate_fields function, so the foreign key validation (which was completely broken, no idea how this wasn't discovered before) lives in a separate function.

File: api/v3/Profile.php [hunk 1]
```diff
@@ -348,12 +348,24 @@ function civicrm_api3_profile_apply($params) {
   );
 
   if (empty($data)) {
-    throw new API_Exception('Enable to format profile parameters.');
+    throw new API_Exception('Unable to format profile parameters.');
```
Comment (eileenmcnaughton) [inline api/v3/Profile.php]: love it....

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
I've reviewed the code changes in this PR and am satisfied that the scope of the changes is limited to the function that provides information when an api call fails and test clean up.

I'm satisfied this is an improvement & is safe to merge provided the tests pass.

**Pull request merged**